### PR TITLE
LPS-148009 [CP 2.0] Create generate new button and route

### PR DESCRIPTION
**Pull request by joanagrego**
PR to create "generate new" button and add route for generate new key form using React Router.

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js [hunk 1]
```diff
@@ -31,26 +31,48 @@ const PAGE_SKELETON_LAYOUT = {
...
 		return currentPath.length
 			? currentPath.slice(-1)[0]
 			: PAGE_TYPES.overview;
-	}, [location]);
+	}, [currentPath]);
+
+	const getCurrentProduct = () => {
+		const activationKey = 'activation';
+
+		const isProduct = !!currentPath?.find((path) => path === activationKey);
```
Comment (igordmf) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:49]: I think you can use ".some" instead "!!.find"

**Review by igordmf — APPROVED** [review]
Good job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js (hunk 1 above)
Comment (iagochaves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:49]: Nice catch

**Review by iagochaves — APPROVED** [review]
Superb work!

**Comment by jairmedeiros** [comment]
@joanagrego, let's try no add links or references to our internal job :). Let's use the JIRA for this

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js (hunk 1 above)
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:49]: @joanagrego

**Review by jairmedeiros — APPROVED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js (hunk 1 above)
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:49]: @joanagrego

**Comment by jairmedeiros** [comment]
Waiting fixes in description and the answer to @igordmf to forward it!
Good job! We are made so many good stuffs here

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js (hunk 1 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js]: Nice! Done!

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Good job, @joanagrego !

**Comment by kevenleone** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:test:relevant

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**